kong/plugins/ldap-auth/asn1.lua

```diff
@@ -1,21 +1,51 @@
+require "resty.openssl.include.ossl_typ"
+local asn1_macro = require "resty.openssl.include.asn1"
+local ffi = require "ffi"
+local C = ffi.C
+local ffi_new = ffi.new
+local ffi_string = ffi.string
+local ffi_cast = ffi.cast
+local assert = assert
+local fmt = string.format
+
 local lpack = require "lua_pack"
 local bpack = lpack.pack
 local bunpack = lpack.unpack
 
-
 local setmetatable = setmetatable
 local tonumber = tonumber
 local reverse = string.reverse
-local ipairs = ipairs
 local concat = table.concat
-local insert = table.insert
 local pairs = pairs
 local math = math
 local type = type
 local char = string.char
 local bit = bit
 
 
+asn1_macro.declare_asn1_functions("ASN1_TYPE")
+asn1_macro.declare_asn1_functions("ASN1_ENUMERATED")
```

Comment:
```suggestion
```
also don't need those two I think, this is useful to populate `X_new`, `X_free` and `X_dup` but we are not using them here.

Reply:
Thanks!